readthedocs/projects/models.py

```diff
@@ -1256,14 +1257,30 @@ def update_stable_version(self):
                 )
                 return new_stable
 
-    def versions_from_branch_name(self, branch):
-        return (
-            self.versions.filter(identifier=branch)
-            | self.versions.filter(identifier="remotes/origin/%s" % branch)
-            | self.versions.filter(identifier="origin/%s" % branch)
-            | self.versions.filter(verbose_name=branch)
+    def versions_from_name(self, name, type=None):
+        """
+        Get all versions attached to the branch or tag name.
+
+        Normally, only one version should be returned, but since LATEST and STABLE
+        are aliases for the branch/tag, they may be returned as well.
+        """
+        queryset = self.versions(manager=INTERNAL)
+        queryset = queryset.filter(
+            # Normal branches
+            Q(verbose_name=name, machine=False)
+            # Latest and stable have the name of the branch/tag in the identifier.
+            # NOTE: if stable is a branch, identifier will be the commit hash,
+            # so we don't have a way to match it by name.
+            # We should do another lookup to get the original stable version,
+            # or change our logic to store the tags name in the identifier of stable.
+            | Q(identifier=name, machine=True)
```

Comment:
Why we were using `remotes/origin/%s` and `origin/%s` previously and we are not now? I'm confused with this method and I'm not sure the logic will be the same. Do we have enough tests for this?

Reply:
We used to save the branches with remote/origin as the identifier, and verbose_name without it, identifier and verbose name are always the same for branches now (this was changed a long time ago when we switched to gitpython...). Anyway, this is always using the verbose name, so even if there are ancient versions that include remote/origin in the identifier, this won't be a breaking change.